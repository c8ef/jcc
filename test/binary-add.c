int foo(void) { return 12 + 30; }
