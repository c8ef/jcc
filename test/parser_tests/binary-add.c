int foo(void) {
  int x = 12 + 30;
  int y = 1 + 2 * 3;
  int z = 10 / 5 - 2;
}
