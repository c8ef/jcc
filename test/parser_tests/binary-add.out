FunctionDecl: foo
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
        BinaryExpr(+):
          IntergerLiteral: 12
          IntergerLiteral: 30
    DeclStatement
      VarDecl: y
        BinaryExpr(+):
          IntergerLiteral: 1
          BinaryExpr(*):
            IntergerLiteral: 2
            IntergerLiteral: 3
    DeclStatement
      VarDecl: z
        BinaryExpr(-):
          BinaryExpr(/):
            IntergerLiteral: 10
            IntergerLiteral: 5
          IntergerLiteral: 2
